src/seat.c

```diff
@@ -352,26 +352,40 @@ focus_change_notify(struct wl_listener *listener, void *data)
 	struct view *view = event->new_surface ?
 		view_from_wlr_surface(event->new_surface) : NULL;
 
+	/* Prevent focus switch to layershell client from updating view state */
+	if (!view && server->active_view && event->new_surface
```

Comment:
It looks like the `server->active_view` condition was copied from the `view_is_related` logic later on? I think it's unnecessary here.

Reply:
Do we actually even need the `!view` in the same place?

Reply:
Probably not

Reply:
Fixed.
Tested with sfwbar... works.